tests/data/dim_keyword.yaml

```diff
@@ -11,4 +11,13 @@ NXarchive(NXobject):
       doc: Testarray with dimensions spelled out
       dimensions:
         rank: 1
-        dim: [[1, 2]]
\ No newline at end of file
+        dim: [[1, 2]]
+    my_dim_with_numpy_3d(NX_NUMBER):
+      dim: (2, 4, 5)
```

Comment:
The `dim` notation with `numpy` comes in action if there are no other dim attributes, no `dimensions` doc, ` dimension` attributes. The numpy only fits with the shape of data not to bring other attributes which are `string`. For string, the use of a list is more conventional. 
But we can add the `numpy` array shape feature in the total block.
e.g. 
```yaml
# 2D particle motion
dimensions:
   rank: 2
   dim: [[0, x coordinate], [1, y coordinate]]
   dim_parameters:
      doc: ["Position of particle on x-axis.","Position of particle on y-axis."]
```
```yaml
# 2D particle motion
dimensions:
   rank: 2
   dim: (
   dim_parameters:
      doc: ["Position of particle on x-axis.","Position of particle on y-axis."]
```

Reply:
> Just for clarification, how would you use the `dim_parameters` if one opts for numpy notation? Like this?
> 
> ```
> my_dim_with_numpy_3d(NX_NUMBER):
>   dim: (2, 4, 5)
>     \@ref: ['s', 'm']
> ```
> 
> If so, please add that case to the tests data.

FYI, I assumed it is a typo. The example you mentioned is not a valid syntax in yaml.

Reply:
No, it was not a typo. I am asking if it is still possible to define dimension parameters when I am using the numpy notation. Let's say, I want to have the following in my nxdl.xml file:
```xml
<field name="my_dim_with_numpy_3d" type="NX_NUMBER">
  <dimensions rank=3>
      <dim index="1" value="2" ref="index_1"/>
      <dim index="2" value="4" ref="index_2"/>
      <dim index="3" value="5" ref="index_3"/>
  </dimensions>
```
Is there anyway to do this with the `dim` keyword or is this only possible with the `dimensions` key word?

For `dimensions`, it would be
```yaml
dimensions:
   rank: 3
   dim: (2, 4, 5)
   dim_parameters:
      doc: ["index_1","index_2", ,"index_3"]
```

I am thinking if it would be useful to add attributes to the numpy-notation so that the same xml could be achieved by e.g.
```yaml
dim: (2, 4, 5)
  \@ref: ["index_1","index_2", ,"index_3"]
```

If this is not supported and/or planned, I am fine with using the longer notatiosn with the `dimension` keyword.

Reply:
Sorry on the last conversation, I mentioned, 
```yaml
dimensions:
   rank: 3
   dim: (2, 4, 5)
   dim_parameters:
      doc: ["index_1","index_2", ,"index_3"]
```
Later, I realised that for `nyaml2nxdl` it is fine. But on the `nxdl2yaml` it follow the old conversion.
```yaml
dimensions:
   rank: 3
   dim: [[1, 500], [2, 100]]
   dim_parameters:
      doc: ["index_1","index_2", ,"index_3"]
```
If we move to the new conversion which makes unnecessary changes on all the `yaml` files. For a few files, it is not an issue. But more than 200 of files it is really overwhelming for us and need to be discussed with all of us. 
So, I prefer to stay on the old style but including the new feature such as 
```yaml
dim: (2,6,7)
```
```xml

Which does not hamper the existing features.

Reply:
Ok, then let's keep it for the `nxdl2yaml` as it is 👍